browser/ui/tabs/split_view_tab_strip_model_adapter.cc

```diff
@@ -265,17 +264,16 @@ void SplitViewTabStripModelAdapter::TabPinnedStateChanged(
   DCHECK(tile->first == source_tab || tile->second == source_tab);
 
   base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
-      FROM_HERE,
-      base::BindOnce(
-          [](base::WeakPtr<SplitViewTabStripModelAdapter> adapter,
-             SplitViewBrowserData::Tile tile, tabs::TabHandle source_tab) {
-            if (!adapter) {
-              return;
-            }
-
-            adapter->SynchronizePinnedState(tile, source_tab);
-          },
-          weak_ptr_factory_.GetWeakPtr(), *tile, source_tab));
+      FROM_HERE, base::BindOnce(
+                     [](base::WeakPtr<SplitViewTabStripModelAdapter> adapter,
+                        TabTile tile, tabs::TabHandle source_tab) {
+                       if (!adapter) {
+                         return;
+                       }
+
+                       adapter->SynchronizePinnedState(tile, source_tab);
```

Comment:
I think this is unsafe because the source tab may be destroyed when this runs

Reply:
also the tabs owned be TabTile may be destroyed

Reply:
you must check `Get()` on `SupportsHandles` to ensure the object the handle points to is still valid (`TabModel`)